ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1216,6 +1216,57 @@ extension SignalType {
 	}
 }
 
+extension SignalType {
+	/// Forwards only those values from `self` that have unique identities across the set of
+	/// all values that have been seen.
+	///
+	/// Note: This causes the identities to be retained to check for uniqueness.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func uniqueValues<Identity: Hashable>(transform: Value -> Identity) -> Signal<Value, Error> {
+		return Signal { observer in
+			let seenValues: Atomic<Set<Identity>> = Atomic([])
+			
+			return self
+				.observe { event in
+					switch event {
+					case let .Next(value):
+						let identity = transform(value)
+						var isUnique: Bool = false
+						seenValues.modify { set in
+							if set.contains(identity) {
+								return set
+							} else {
+								isUnique = true
+								var mutableSet = set
+								mutableSet.insert(identity)
+								return mutableSet
+							}
+						}
+						if isUnique {
+							fallthrough
+						}
+						
+					default:
```

Comment:
`case .Failed, .Completed, .Interrupted:` would be preferable as with `skipWhile`.